index.js

```diff
@@ -23,22 +24,44 @@ var plugin = module.exports = function (opts) {
   });
   opts.includeUntested = opts.includeUntested === true;
 
-  var instrumenter = new opts.instrumenter(opts);
+  var defaultInstrumenter = new opts.instrumenter(opts);
 
   return through(function (file, enc, cb) {
+    var instrumenter;
+    var fileContents = file.contents.toString();
+    if (file.sourceMap) {
+      var fileOpts = _.defaultsDeep(_.cloneDeep(opts), {
+        codeGenerationOptions: {
+          sourceMap: file.sourceMap.file,
+          sourceMapWithCode: true,
+          sourceContent: fileContents,
+          sourceMapRoot: file.sourceMap.sourceRoot,
+          file: file.path
+        }
+      });
+      instrumenter = new opts.instrumenter(fileOpts);
```

Comment:
Before my changes, the instrumenter was created once for all files in the stream. My reasoning was that creating it might potentially be expensive. Thus, I wanted to only create a new instrumenter for each file if it’s really necessary (because we need to change its options).

Would you prefer something like this?:

```javascript
var fileOpts = opts;
if (file.sourceMap) {
  fileOpts = _.defaultsDeep(_.cloneDeep(opts), {
    codeGenerationOptions: {
      sourceMap: file.sourceMap.file,
      sourceMapWithCode: true,
      sourceContent: fileContents,
      sourceMapRoot: file.sourceMap.sourceRoot,
      file: file.path
    }
});
var instrumenter = new opts.instrumenter(fileOpts);
```